pandas/core/indexes/range.py

```diff
@@ -216,39 +216,69 @@ def _format_data(self, name=None):
         return None
 
     # --------------------------------------------------------------------
-    @property
+    @cache_readonly
     def start(self):
         """
         The value of the `start` parameter (or ``0`` if this was not supplied)
         """
         # GH 25710
-        return self._start
+        return self._range.start
 
     @property
+    def _start(self):
+        """
+        The value of the `start` parameter (or ``0`` if this was not supplied)
+
+         .. deprecated:: 0.25.0
+            Use ._range.start or .start instead.
+        """
+        return self._range.start
```

Comment:
Unfortunately, fast_parquet uses ``_start`` etc. which I hadn't predicted, when I started this PR. So this proposes deprecating the attributes ``_start``, ``_stop`` and ``_step``, and removing them in 1.0.

This can be bit of a nuisance for upstream maintainers, so don't know if there's opposition to this, but this PR has several benefits as mentioned, so there's a trade-off.

Reply:
I think pyarrow uses them as well. IIRC recently had a discussion that they're semi-officially public for advanced users (do you recall that @jorisvandenbossche)

Reply:
don't recommend ._range.start, just say .start

Reply:
I think its ok to deprecate these, the downstream will have to adjust